source/common/tracing/http_tracer_manager_impl.cc

```diff
@@ -18,11 +18,19 @@ HttpTracerManagerImpl::getOrCreateHttpTracer(const envoy::config::trace::v3::Tra
   const auto cache_key = MessageUtil::hash(*config);
   const auto it = http_tracers_.find(cache_key);
   if (it != http_tracers_.end()) {
-    return it->second;
+    auto http_tracer = it->second.lock();
+    if (http_tracer) { // HttpTracer might have been released since it's a weak reference
+      return http_tracer;
+    } else {
+      http_tracers_.erase(it);
+    }
```

Comment:
Given that you walk the list in the next line, I would probably delete this since it will get cleaned anyway below.

Reply:
removed